src/unit_tests/ctx/sync_library_with_api.rs

```diff
@@ -295,3 +295,251 @@ fn actionctx_synclibrarywithapi_with_user() {
+                        (
+                            LOCAL_NOT_REMOVED_NOT_WATCHED_ITEM.id.to_owned(),
+                            LOCAL_NOT_REMOVED_NOT_WATCHED_ITEM.to_owned(),
+                        ),
+                    ]
+                    .into_iter()
+                    .collect(),
+                },
+                ..Default::default()
+            },
+        },
+        1000,
+    );
+    run(runtime.dispatch(&Msg::Action(Action::Ctx(ActionCtx::SyncLibraryWithAPI))));
+    assert_eq!(
+        runtime.app.read().unwrap().ctx.library,
+        LibBucket {
+            uid: Some("user_id".to_string()),
+            items: vec![
+                (
+                    LOCAL_REMOVED_NOT_WATCHED_ITEM.id.to_owned(),
+                    LOCAL_REMOVED_NOT_WATCHED_ITEM.to_owned(),
+                ),
+                (
+                    LOCAL_REMOVED_WATCHED_ITEM.id.to_owned(),
+                    LOCAL_REMOVED_WATCHED_ITEM.to_owned(),
+                ),
+                (
+                    LOCAL_NOT_REMOVED_NOT_WATCHED_ITEM.id.to_owned(),
+                    LOCAL_NOT_REMOVED_NOT_WATCHED_ITEM.to_owned(),
+                ),
+                (
+                    LOCAL_NOT_REMOVED_WATCHED_ITEM.id.to_owned(),
+                    LOCAL_NOT_REMOVED_WATCHED_ITEM.to_owned(),
+                ),
+            ]
+            .into_iter()
+            .collect(),
+        },
+        "library updated successfully in memory"
+    );
+    assert!(
+        STORAGE
+            .read()
+            .unwrap()
+            .get(LIBRARY_RECENT_STORAGE_KEY)
+            .map(|data| serde_json::from_str::<(UID, Vec<LibItem>)>(&data).unwrap())
+            .map_or(false, |(uid, items)| {
+                uid == Some("user_id".to_owned())
+                    && items.len() == 4
+                    && items.contains(&LOCAL_REMOVED_NOT_WATCHED_ITEM)
+                    && items.contains(&LOCAL_REMOVED_WATCHED_ITEM)
+                    && items.contains(&LOCAL_NOT_REMOVED_NOT_WATCHED_ITEM)
+                    && items.contains(&LOCAL_NOT_REMOVED_WATCHED_ITEM)
+            }),
+        "Library recent slot updated successfully in storage"
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().len(),
+        3,
+        "Three requests have been sent"
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().get(0).unwrap().to_owned(),
+        Request {
+            url: "https://api.strem.io/api/datastoreMeta".to_owned(),
+            method: "POST".to_owned(),
+            body: "{\"authKey\":\"auth_key\",\"collection\":\"libraryItem\"}".to_owned(),
+            ..Default::default()
+        },
+        "datastoreMeta request has been sent"
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().get(1).unwrap().url,
+        "https://api.strem.io/api/datastorePut".to_owned(),
+        "datastorePut request has been sent"
+    );
+    assert_eq!(
+        REQUESTS.read().unwrap().get(2).unwrap().to_owned(),
```

Comment:
this one is unnecessary. havent thought about that while writing the sync logic.
if there are no items to be pulled/pushed no requests should be sent.